### PR TITLE
Migration to circle sdk

### PR DESCRIPTION
**Pull request by avkos**

### CODE REVIEW

File: packages/circle-demo-webapp/app/components/ChainLabel/ChainLabel.tsx [hunk 1]
```diff
@@ -1,43 +1,44 @@
 import { NetworkIcon } from '@web3icons/react';
-import { BLOCKCHAIN } from 'web3-circle-sdk';
+
+import { Blockchain } from '~/lib/constants';
 
 export interface ChainSelectProps {
   /** The blockchain network */
-  blockchain: BLOCKCHAIN;
+  blockchain: string;
```
Comment (danforbes) [inline packages/circle-demo-webapp/app/components/ChainLabel/ChainLabel.tsx:7]: Why is this being changed from the enum to a string?

File: packages/circle-demo-webapp/app/components/TransactionTableRow/TransactionTableRow.stories.ts [hunk 2]
```diff
@@ -20,14 +22,14 @@ export const Default: Story = {
...
-      blockchain: 'MATIC-AMOY' as BLOCKCHAIN,
+      blockchain: Blockchain.MaticAmoy,
       tokenId: '36b6931a-873a-56a8-8a27-b706b17104ee',
       walletId: '24d1ad14-cf0c-5d7d-96d1-aca6447d0fdc',
       sourceAddress: '0x2da2bf0a07b015ffa80821df8b2203d473964d95',
       destinationAddress: '0xf6c9efc84080217ccd13ef6d4a7f26a680f2c713',
       transactionType: 'INBOUND',
       custodyType: 'DEVELOPER',
-      state: 'COMPLETE' as TRANSACTION_STATE,
+      state: 'COMPLETE' as TransactionState,
```
Comment (danforbes) [inline packages/circle-demo-webapp/app/components/TransactionTableRow/TransactionTableRow.stories.ts:32]: Why not use `TransactionState.Complete` like you did for `TransactionDetails`?

File: packages/circle-demo-webapp/app/components/WalletBalance/WalletBalance.tsx [hunk 3]
```diff
@@ -1,8 +1,8 @@
 import { TokenIcon } from '@web3icons/react';
-import type { WalletTokenBalance } from 'web3-circle-sdk';
+
+import { WalletTokenBalance } from '~/lib/types';
 
 export interface WalletBalanceProps {
-  /** The balance details */
   balance: WalletTokenBalance;
```
Comment (danforbes) [inline packages/circle-demo-webapp/app/components/WalletBalance/WalletBalance.tsx:6]: Why did you remove the comment?

File: packages/circle-demo-webapp/app/routes/wallets.$id/route.tsx [hunk 4]
```diff
@@ -1,11 +1,12 @@
 import { ActionFunctionArgs } from '@remix-run/node';
 import { Link, useLoaderData, useParams } from '@remix-run/react';
-import type { BLOCKCHAIN, WalletSet } from 'web3-circle-sdk';
+import type { WalletSet } from 'web3-circle-sdk';
```
Comment (danforbes) [inline packages/circle-demo-webapp/app/routes/wallets.$id/route.tsx:3]: Should this be coming from `~/lib/types`?

File: packages/circle-demo-webapp/app/lib/types.ts
Comment (Muhammad-Altabba) [inline packages/circle-demo-webapp/app/lib/types.ts:1]: We can suggest to later add those to Cirlce SDKs. 
As I am wondering why they have it inside their demo app https://github.com/circlefin/w3s-sample-user-controlled-client-web/blob/c504df2144bc5c969a3aa5e5002c6672ef7ea22c/app/shared/types.ts#L136 and not inside the SDK. We just need to check this later.

File: packages/circle-demo-webapp/app/lib/types.ts [hunk 5]
```diff
@@ -0,0 +1,757 @@
...
+}
+
+/**
+ * A developer controlled wallet set
+ * https://developers.circle.com/api-reference/w3s/developer-controlled-wallets/create-wallet-set
+ * https://developers.circle.com/api-reference/w3s/developer-controlled-wallets/update-wallet-set
+ * https://developers.circle.com/api-reference/w3s/developer-controlled-wallets/get-wallet-sets
+ * https://developers.circle.com/api-reference/w3s/developer-controlled-wallets/get-wallet-set
+ */
+export interface WalletSet {
```
Comment (krzysu) [inline packages/circle-demo-webapp/app/lib/types.ts:36]: wait, doesn't circle sdk export type definitions? I will check what they have then.

File: packages/circle-demo-webapp/app/routes/wallets.$id/route.tsx [hunk 6]
```diff
@@ -16,25 +17,29 @@ export async function loader({ params }: { params: { id: string } }) {
...
+  const [walletsResp, walletSetResp] = await Promise.all([
+    sdk.listWallets({ walletSetId: id }),
+    sdk.getWalletSet({ id }),
   ]);
 
   return {
-    wallets,
-    walletSet,
+    wallets: walletsResp?.data?.wallets,
+    walletSet: walletSetResp?.data?.walletSet,
```
Comment (Muhammad-Altabba) [inline packages/circle-demo-webapp/app/routes/wallets.$id/route.tsx:27]: Probably casting here instead of casting inside the component?
```suggestion
    wallets: walletsResp?.data?.wallets as Wallet[],
    walletSet: walletSetResp?.data?.walletSet as WalletSet,
```

**Review by Muhammad-Altabba — APPROVED** [review]
All sounds great to me. Except for Dan's comment and another 2 tiny comments.

File: packages/circle-demo-webapp/app/routes/wallets.$id/route.tsx (hunk 4 above)
Comment (avkos) [reply packages/circle-demo-webapp/app/routes/wallets.$id/route.tsx:3]: typo. fixed

File: packages/circle-demo-webapp/app/components/WalletBalance/WalletBalance.tsx (hunk 3 above)
Comment (avkos) [reply packages/circle-demo-webapp/app/components/WalletBalance/WalletBalance.tsx:6]: fixed

File: packages/circle-demo-webapp/app/lib/types.ts (hunk 5 above)
Comment (avkos) [reply packages/circle-demo-webapp/app/lib/types.ts:36]: unfortunately, they do not export types.

File: packages/circle-demo-webapp/app/components/TransactionTableRow/TransactionTableRow.stories.ts (hunk 2 above)
Comment (avkos) [reply packages/circle-demo-webapp/app/components/TransactionTableRow/TransactionTableRow.stories.ts:32]: fixed

**Review by danforbes — APPROVED** [review]

**Review by krzysu — APPROVED** [review]
